### PR TITLE
Adding a key for quick VR/normal toggle

### LINKED ISSUE #124: add a keyboard shortcut for entering/exiting VR mode
I'd recommend a keypress of the `v` key (without any alt/shift/control/meta keys held).

### PR DESCRIPTION
**Pull request by delapuente**
Fix #124

### CODE REVIEW

**Comment by caseyyee** [comment]
I thought that on vrdisplayactivate event should deal with the case of easily entering VR?  Should be just a simple case of putting on your headset.   I don't think this actually works though, not sure if that's a browser issue?

Also, maybe we also need to make a decision on if we want to handle interactions entirely in unity or not.  I think being consistent here will be important.   If we start binding keypresses in Javascript, then we have the chance to interfering/overlapping with Unity keypresses as well.

Would rather not bind keys if we don't have to.

@cvan?

**Comment by delapuente** [comment]
> I thought that on vrdisplayactivate event should deal with the case of easily entering VR?

Sorry, what do you mean?

> Also, maybe we also need to make a decision on if we want to handle interactions entirely in unity or not.

Unfortunately, this is not possible when entering VR because it requires a user gesture. So, for this specific case, you need to enter via JS. You have an explanation [in the commit comment](https://github.com/mozilla/unity-webvr-export/pull/144/commits/e0f41c9223348c3cbd5faddb0322de084e288768).

File: Assets/WebVR/Plugins/WebGL/webvr.jslib [hunk 1]
```diff
@@ -10,5 +10,9 @@ mergeInto(LibraryManager.library, {
 
   PostRender: function () {
     document.dispatchEvent(new CustomEvent('Unity', {detail: 'PostRender'}));
+  },
+
+  ConfigureToggleVRKeyName: function (keyName) {
+    document.dispatchEvent(new CustomEvent('Unity', {detail: 'ConfigureToggleVRKeyName:'+Pointer_stringify(keyName)}));
```
Comment (cvan) [inline Assets/WebVR/Plugins/WebGL/webvr.jslib]: nit: add spaces around `+`

File: Assets/WebGLTemplates/WebVR/webvr.js [hunk 2]
```diff
@@ -317,13 +323,20 @@
     });
   }
 
+  function onKeyUp(evt) {
+    if (toggleVRKeyName && toggleVRKeyName === evt.key) {
```
Comment (cvan) [inline Assets/WebGLTemplates/WebVR/webvr.js]: which default key is this?

Comment (cvan) [reply Assets/WebGLTemplates/WebVR/webvr.js]: and we should look at `altKey`, `ctrlKey`, `metaKey`, right?

**Comment by caseyyee** [comment]
r+wc

File: Assets/WebVR/Scenes/WebVR.unity [hunk 3]
```diff
@@ -236,101 +236,6 @@ Transform:
   m_Father: {fileID: 1807604226}
   m_RootOrder: 0
   m_LocalEulerAnglesHint: {x: 143.729, y: -15.052002, z: 46.773987}
---- !u!1 &143091789
```
Comment (cvan) [inline Assets/WebVR/Scenes/WebVR.unity]: are these changes to this file related?

File: Assets/WebVR/Scripts/WebVRCamera.cs [hunk 4]
```diff
@@ -39,9 +42,14 @@ public class WebVRCamera : MonoBehaviour
...
     
-	public GameObject leftHandObj;
-    public GameObject rightHandObj;
+	[Tooltip("GameObject to be controlled by the left hand controller.")]
+	public GameObject leftHandObject;
+    
+	[Tooltip("GameObject to be controlled by the right hand controller.")]
+	public GameObject rightHandObject;
 
+ 	[Tooltip("Name of the key used to alternate between VR and normal mode. Leave in blank to disable.")]
```
Comment (cvan) [inline Assets/WebVR/Scripts/WebVRCamera.cs]: not: `Leave in blank` -> `Leave blank`

File: Assets/WebVR/Prefabs/WebVRCameraSet.prefab [hunk 5]
```diff
@@ -284,8 +284,9 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: 2330d5cea21bb564e97ad081f15487d9, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
-  leftHandObj: {fileID: 0}
-  rightHandObj: {fileID: 0}
+  leftHandObject: {fileID: 0}
+  rightHandObject: {fileID: 0}
+  toggleVRKeyName: v
```
Comment (cvan) [inline Assets/WebVR/Prefabs/WebVRCameraSet.prefab]: :+1: good work on this being a configurable option

Comment (caseyyee) [reply Assets/WebVR/Prefabs/WebVRCameraSet.prefab]: Nice, I like that

File: Assets/WebGLTemplates/WebVR/webvr.js (hunk 2 above)
Comment (delapuente) [reply Assets/WebGLTemplates/WebVR/webvr.js]: > and we should look at altKey, ctrlKey, metaKey, right?

Right, we should do it as part of #8

File: Assets/WebVR/Scenes/WebVR.unity (hunk 3 above)
Comment (delapuente) [reply Assets/WebVR/Scenes/WebVR.unity]: Yes, it updates the scene to use the modified prefab. Don't ask me how Unity does it ;)

File: Assets/WebGLTemplates/WebVR/webvr.js (hunk 2 above)
Comment (cvan) [reply Assets/WebGLTemplates/WebVR/webvr.js]: yeah, per your comments in last meeting, I now understand how you want to approach this. I think that’s a good strategy. (please ignore my comment for now.)

File: Assets/WebVR/Scenes/WebVR.unity (hunk 3 above)
Comment (cvan) [reply Assets/WebVR/Scenes/WebVR.unity]: I am in no position to question that, heh :)

File: Assets/WebVR/Scripts/WebVRCamera.cs [hunk 6]
```diff
@@ -143,9 +151,12 @@ public void End()
 	}
 
 	private void toggleMode() {
-		active = active == true ? false : true;
-		string mode = active == true ? "vr" : "normal";
+		active = !active;
+		string mode = active ? "vr" : "normal";
+
+		#if UNITY_EDITOR || !UNITY_WEBGL
 		changeMode (mode);
```
Comment (cvan) [inline Assets/WebVR/Scripts/WebVRCamera.cs]: major nit: not from you, but I noticed the style (CamelCase vs camelBack) is different here

Comment (delapuente) [reply Assets/WebVR/Scripts/WebVRCamera.cs]: Moving to #104

**Pull request merged**